src/jvlink/constants.py

```
@@ -387,6 +406,26 @@ def _split_jvopen_data_specs(data_spec: str) -> tuple[str, ...]:
     return tuple(data_spec[index:index + 4] for index in range(0, len(data_spec), 4))
 
 
+def jvopen_supports_end_timestamp(data_spec: str) -> bool:
+    """Return True iff every component may take the official start-end form.
+
+    公式仕様 p.17-18 のとおり、終了時刻禁止 ID をひとつでも含む dataspec に
+    終了ポイント時刻を付けると -1 が返り「該当データなし」と区別できない。
+    そのため全構成要素が許す場合に限り True。不正・空の dataspec は False
+    （開始のみ形式へ倒す）とし、そもそも validate_jvopen_combination が先に
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)